src/server/engine_shard_set.cc

```diff
@@ -72,37 +88,44 @@ EngineShard::Stats& EngineShard::Stats::operator+=(const EngineShard::Stats& o)
   return *this;
 }
 
+void EngineShard::DefragTaskState::UpdateState(uint64_t cursor_val) {
+  cursor = cursor_val;
+  if (cursor == kCursorDoneState) {
```

Comment:
no need this check and this explanation. you can just conditionalessly set `underutilized_found = false;`

Reply:
changed (as well as the function name).